integrations/axum/src/lib.rs

```diff
@@ -110,20 +110,54 @@ pub fn redirect(cx: leptos::Scope, path: &str) {
 }
 
 /// Decomposes an HTTP request into its parts, allowing you to read its headers
-/// and other data without consuming the body.
-pub async fn generate_request_parts(req: Request<Body>) -> RequestParts {
+/// and other data without consuming the body. Creates a new Request from the
+/// original parts for further processsing
+pub async fn generate_request_parts(
+    req: Request<Body>,
+) -> (Request<Body>, RequestParts) {
     // provide request headers as context in server scope
     let (parts, body) = req.into_parts();
     let body = body::to_bytes(body).await.unwrap_or_default();
-    RequestParts {
-        method: parts.method,
-        uri: parts.uri,
-        headers: parts.headers,
+    let request_parts = RequestParts {
+        method: parts.method.clone(),
+        uri: parts.uri.clone(),
+        headers: parts.headers.clone(),
         version: parts.version,
-        body,
-    }
+        body: body.clone(),
+    };
+    let request = Request::from_parts(parts, body.into());
+
+    (request, request_parts)
 }
 
+/// A struct to hold the http::request::Request and allow users to take ownership of it
+/// Requred by Request not being Clone. See this issue for eventual resolution: https://github.com/hyperium/http/pull/574
+#[derive(Debug, Default)]
+pub struct LeptosRequest<B>(Arc<RwLock<Option<Request<B>>>>);
+
+impl<B> Clone for LeptosRequest<B> {
+    fn clone(&self) -> Self {
+        Self(self.0.clone())
+    }
+}
+impl<B> LeptosRequest<B> {
```

Comment:
`LeptosRequest` probably also needs some kind of `.take_request()` method or something, right? I think because it's a private inner field there's not any way to get the request out at present.

Reply:
Yeeah, accidentally didn't commit that, fixed.